src/ScrollViewGesture.tsx

```diff
@@ -285,89 +280,184 @@ const IScrollViewGesture: React.FC<PropsWithChildren<Props>> = (props) => {
+    (e: PanGestureHandlerEventPayload) => {
+      'worklet';
 
-    if (validStart.value) {
-      validStart.value = false;
-      cancelAnimation(translation);
-    }
-    touching.value = true;
-    const { translationX, translationY } = e;
-    const panTranslation = isHorizontal.value
-      ? translationX
-      : translationY;
-    if (!infinite) {
-      if ((translation.value > 0 || translation.value < -max.value)) {
-        const boundary = translation.value > 0 ? 0 : -max.value;
-        const fixed = boundary - panOffset.value;
-        const dynamic = panTranslation - fixed;
-        translation.value = boundary + dynamic * 0.5;
-        return;
+      if (validStart.value) {
+        validStart.value = false;
+        cancelAnimation(translation);
+      }
+      touching.value = true;
+      const { translationX, translationY } = e;
+      const panTranslation = isHorizontal.value ? translationX : translationY;
+      if (!infinite) {
+        if (translation.value > 0 || translation.value < -max.value) {
+          const boundary = translation.value > 0 ? 0 : -max.value;
+          const fixed = boundary - panOffset.value;
+          const dynamic = panTranslation - fixed;
+          translation.value = boundary + dynamic * 0.5;
+          return;
+        }
+      }
+
+      const translationValue = panOffset.value + panTranslation;
+      translation.value = translationValue;
+    },
+    [isHorizontal, max, panOffset, infinite, overscrollEnabled, translation, validStart, touching],
+  );
+
+  const onGestureFinish = useCallback(
+    (e: GestureStateChangeEvent<PanGestureHandlerEventPayload>) => {
+      'worklet';
+
+      const { velocityX, velocityY, translationX, translationY } = e;
+      scrollEndVelocity.value = isHorizontal.value ? velocityX : velocityY;
+      scrollEndTranslation.value = isHorizontal.value ? translationX : translationY;
+
+      const totalTranslation = scrollEndVelocity.value + scrollEndTranslation.value;
+
+      if (
+        maxScrollDistancePerSwipeIsSet &&
+        Math.abs(totalTranslation) > maxScrollDistancePerSwipe
+      ) {
+        const nextPage =
+          Math.round(
+            (panOffset.value + maxScrollDistancePerSwipe * Math.sign(totalTranslation)) / size,
+          ) * size;
+        translation.value = withSpring(withProcessTranslation(nextPage), onScrollEnd);
+      } else {
+        endWithSpring(onScrollEnd);
       }
+
+      if (!infinite) touching.value = false;
+    },
+    [
+      size,
+      infinite,
+      touching,
+      panOffset,
+      translation,
+      isHorizontal,
+      scrollEndVelocity,
+      scrollEndTranslation,
+      maxScrollDistancePerSwipeIsSet,
+      maxScrollDistancePerSwipe,
+      endWithSpring,
+      withSpring,
+      onScrollEnd,
+    ],
+  );
+
+  const gesture = useMemo(() => {
+    const panGesture = Gesture.Pan()
+      .onBegin(onGestureBegin)
+      .onUpdate(onGestureUpdate)
+      .onEnd(onGestureFinish);
+
+    const {
```

Comment:
destructure panGestureHandlerProps, we need to handle each separately, we cannot iterate over them and call the functions on Gesture.Pan() programmatically because they do not match the prop names.